modules/bvl_feedback/ajax/bvl_panel_ajax.php

```diff
@@ -15,25 +15,30 @@
  * @link     https://www.github.com/aces/Loris-Trunk/
  */
 
-$user     =& User::singleton();
-$username = $user->getUsername();
+namespace LORIS\bvl_feedback;
+use \LORIS\StudyEntities\Candidate\CandID;
+
+$username = \User::singleton()->getUsername();
 
 if (isset($_POST['candID']) && (empty($_POST['sessionID']))) {
-    $feedbackThread =& NDB_BVL_Feedback::Singleton($username, $_POST['candID']);
+    $candID         = new CandID($_POST['candID']);
```

Comment:
`$_POST['candID']` not sanitized.
Did some code disappeared in a rebase?

Reply:
The CandID class automatically throws an exception when the string input is not numeric and is not exactly 6 characters in length. I don't believe it's possible to create a malicious JS injection attack using numbers so this input can be considered safe.

Reply:
In that case, I will clear my opposition and approve this PR.